packages/schematics/angular/migrations/update-vscode-recommendations/ast-utils.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Rule, Tree } from '@angular-devkit/schematics';
```

Comment:
seems kinda crazy to me that schematics doesn't have some reusable helper for the common case of mutating some json.

Aside, have you seen https://stedolan.github.io/jq/ ? it's an amazing command-line based tool that gives you unix sed-style power over JSON

Reply:
We have a lot of these kinds of useful helpers in our nx repo to help with making cli schematics. Maybe someday we can move some of the better utilities upstream if you think they would be handy

Reply:
jq looks really cool I think I'm going to start looking into using that more in the future. We've been trying to get values from JSONs on disk without parsing them and that might do the trick

Reply:
I started hooking jq into the bazel pipeline at one point, works nicely as an independant action. maybe should do that again and land it...

Reply:
The upcoming workspace writer feature requires non-destructive (retention of comments/formatting/etc) JSON5 mutation support.  Once that lands, the underlying components can then be generalized and moved to an additional Public API for scenarios such as this.